### PR TITLE
update warmup lr example

### LINKED ISSUE #9600: warmup example in the document conflicts with lr scheduler and lr monitor
### To Reproduce

The warm-up example in the [document](https://pytorch-lightning.readthedocs.io/en/latest/common/lightning_module.html#optimizer-step) suggests to manually warm-up lr before optimizer step. 
```
# learning rate warm-up
def optimizer_step(self, current_epoch, batch_nb, optimizer, optimizer_idx, second_order_closure=None, on_tpu=False, using_native_amp=False, using_lbfgs=False):
    # warm up lr
    if self.trainer.global_step < 500:
        lr_scale = min(1., float(self.trainer.global_step + 1) / 500.)
        for pg in optimizer.param_groups:
            pg['lr'] = lr_scale * self.hparams.learning_rate

    # update params
    optimizer.step()
    optimizer.zero_grad()
```

However, in the `training_epoch_loop.py`, the scheduler updater will change the lr. 
https://github.com/PyTorchLightning/pytorch-lightning/blob/381343a79c703f2ccf1ab7c1d87400ad6e31fdf4/pytorch_lightning/loops/epoch/training_epoch_loop.py#L137-L150

I think this will not affect the warm-up behavior of optimizer, since the updated lr is never used.
But, will it affect the lr monitor? According to the [documentation](https://pytorch-lightning.readthedocs.io/en/latest/api/pytorch_lightning.callbacks.lr_monitor.html?highlight=lr), the monitor is call on the start of the batch/epoch. 

**So during warm-up stage, a wrong lr is recorded at the beginning of the next batch-loop.**

### Expected behavior

1. Correct warm-up learning-rate to be recorded.
2. Suggestion for better warmp-up code style. Update documentation.

### Environment

- PyTorch Lightning Version (1.4.5)

### Additional context

Related issue https://github.com/PyTorchLightning/pytorch-lightning/issues/4902

### PR DESCRIPTION
**Pull request by rohitgr7**
## What does this PR do?

Fixes #9600
Also removed the tbptt example since it's already present [here](https://pytorch-lightning.readthedocs.io/en/latest/guides/data.html#truncated-backpropagation-through-time-tbptt).

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [ ] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?
- [ ] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @borda

### CODE REVIEW

**Review by tchaton — APPROVED** [review]
LGTM ! Wonder if we want to promote this example. Doesn't look very composable and a scheduler might be better.

**Review by awaelchli — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]
